hack/minikube-rook-setup.sh

```diff
@@ -0,0 +1,131 @@
+set -x
+
+if [[ $1 == "delete" ]]
+then
+	minikube delete --profile="${PROFILE}"
+	virsh vol-delete --pool "${POOL_NAME}" "${IMAGE_NAME}-${PROFILE}"
+    exit 0
+fi
+
+if [[ $1 == "stop" ]]; then
+	minikube stop --profile="${PROFILE}"
+    exit 0
+fi
+
+if [[ $1 == "start" ]]; then
+	sudo virsh start "${PROFILE}"
+	wait_for_ssh "$PROFILE"
+	minikube ssh "sudo mkdir -p /mnt/vda1/rook/ && sudo ln -sf /mnt/vda1/rook/ /var/lib/rook" --profile="${PROFILE}"
+	minikube start --profile="${PROFILE}"
+    exit 0
+fi
+
+if [[ $1 != "create" ]]; then
+	usage
+fi
+
+### $1 == "create"
+# TODO: Check if already created and bail out!
+## Preserve /var/lib/rook into existing vda1 disk ##
+# TODO: minikube instance is assumed to be kvm2 based, check it?
+minikube ssh "sudo mkdir -p /mnt/vda1/rook/ && sudo ln -sf /mnt/vda1/rook/ /var/lib/rook" --profile="${PROFILE}"
+
+## Create and attach an OSD disk for Ceph ##
+if [[ $(virsh pool-list | grep -wc "${POOL_NAME}") == 0 ]]; then
+	virsh pool-create-as --name "${POOL_NAME}" --type dir --target "${IMAGE_DIR}"
+fi
+if [ ! -f "${IMAGE_DIR}/${IMAGE_NAME}-${PROFILE}" ]; then
+	virsh vol-create-as --pool "${POOL_NAME}" --name "${IMAGE_NAME}-${PROFILE}" --capacity 32G --format qcow2
+fi
+if [[ $(minikube ssh 'echo 1 | sudo tee /sys/bus/pci/rescan > /dev/null ; dmesg | grep virtio_blk' --profile="${PROFILE}" | grep -wc vdb) == 0 ]]; then
+	sudo virsh attach-disk --domain "${PROFILE}" --source "${IMAGE_DIR}/${IMAGE_NAME}-${PROFILE}" --target vdb --persistent --driver qemu --subdriver qcow2 --targetbus virtio
+fi
+minikube ssh 'echo 1 | sudo tee /sys/bus/pci/rescan > /dev/null ; dmesg | grep virtio_blk' --profile="${PROFILE}"
+
+## Install rook-ceph ##
+kubectl apply -f "${ROOK_SRC}/common.yaml" --context="${PROFILE}"
+kubectl apply -f "${ROOK_SRC}/crds.yaml" --context="${PROFILE}"
+# Fetch operator.yaml and enable CSI volume replication
+TMP_OPERATOR_YAML=$(mktemp --suffix .yaml)
+wget "${ROOK_SRC}/operator.yaml" -O "${TMP_OPERATOR_YAML}"
+sed -e 's,CSI_ENABLE_VOLUME_REPLICATION: "false",CSI_ENABLE_VOLUME_REPLICATION: "true",' -i "${TMP_OPERATOR_YAML}"
+sed -e 's,# CSI_VOLUME_REPLICATION_IMAGE: "quay.io/csiaddons/volumereplication-operator:v0.1.0",CSI_VOLUME_REPLICATION_IMAGE: "quay.io/csiaddons/volumereplication-operator:latest",' -i "${TMP_OPERATOR_YAML}"
+sed -e 's,# CSI_ENABLE_OMAP_GENERATOR: "false",CSI_ENABLE_OMAP_GENERATOR: "true",' -i "${TMP_OPERATOR_YAML}"
+kubectl apply -f "${TMP_OPERATOR_YAML}" --context="${PROFILE}"
+rm -f "${TMP_OPERATOR_YAML}"
+# Create a dev ceph cluster
+kubectl apply -f "${scriptdir}/dev-rook-cluster.yaml" --context="${PROFILE}"
+kubectl apply -f "${ROOK_SRC}/toolbox.yaml" --context="${PROFILE}"
+
+# Create a mirroring enabled RBD pool
+kubectl apply -f "${scriptdir}/dev-rook-rbdpool.yaml" --context="${PROFILE}"
+
+# Ensure the pool is created and ready
+wait_for_condition "Ready" kubectl get cephblockpool -n rook-ceph replicapool -o jsonpath='{.status.phase}' --context="${PROFILE}"
```

Comment:
This line has failed the last two Thursdays.  For example:

```
++ kubectl get cephblockpool -n rook-ceph replicapool -o 'jsonpath={.status.phase}' --context=hub
+ result=
+ [[ '' == \R\e\a\d\y ]]
+ count=0
+ sleep 5
+ (( count > 0 ))
+ echo 'Failed to meet Ready for command kubectl get cephblockpool -n rook-ceph replicapool -o jsonpath={.status.phase} --context=hub'
Failed to meet Ready for command kubectl get cephblockpool -n rook-ceph replicapool -o jsonpath={.status.phase} --context=hub
+ exit 1
```

It seems to correspond with a new rook image release:

6/3 v1.6.4
https://github.com/rook/rook/pull/8053

```
$ kubectl --context hub -n rook-ceph describe pods/rook-ceph-operator-65b84f48f8-7mjnj
Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  21m                 default-scheduler  Successfully assigned rook-ceph/rook-ceph-operator-65b84f48f8-7mjnj to hub
  Normal   Pulling    19m (x4 over 21m)   kubelet            Pulling image "rook/ceph:v1.6.4"
  Warning  Failed     19m (x4 over 21m)   kubelet            Failed to pull image "rook/ceph:v1.6.4": rpc error: code = Unknown desc = Error response from daemon: manifest for rook/ceph:v1.6.4 not found: manifest unknown: manifest unknown
  Warning  Failed     19m (x4 over 21m)   kubelet            Error: ErrImagePull
  Normal   BackOff    16m (x20 over 21m)  kubelet            Back-off pulling image "rook/ceph:v1.6.4"
  Warning  Failed     61s (x86 over 21m)  kubelet            Error: ImagePullBackOff
  ```

6/10 v1.6.5
https://github.com/rook/rook/pull/8111

```
$ kubectl --context hub -n rook-ceph describe pods/rook-ceph-operator-5b94b79
f5-wkhfs
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  48m                    default-scheduler  Successfully assigned rook-ceph/rook-ceph-operator-5b94b79f5-wkhfs to hub
  Normal   Pulling    46m (x4 over 48m)      kubelet            Pulling image "rook/ceph:v1.6.5"
  Warning  Failed     46m (x4 over 48m)      kubelet            Failed to pull image "rook/ceph:v1.6.5": rpc error: code = Unknown desc = Error response from daemon: manifest for rook/ceph:v1.6.5 not found: manifest unknown: manifest unknown
  Warning  Failed     46m (x4 over 48m)      kubelet            Error: ErrImagePull
  Warning  Failed     17m (x131 over 48m)    kubelet            Error: ImagePullBackOff
  Normal   BackOff    2m57s (x195 over 48m)  kubelet            Back-off pulling image "rook/ceph:v1.6.5"
  ```

Each time the issue was resolved later in the day, but not before the script gave up waiting and failed.  I wonder if this is to be expected with a new release or a bug each time.  If it is to be expected, one potential solution is to specify a rook version number somehow, perhaps in `hack/dev-rook-cluster.yaml`.

Reply:
One way to set a version is to change line 14 to get `https://raw.githubusercontent.com/rook/rook/v1.6.5/cluster/examples/kubernetes/ceph/operator.yaml`, IOW a specific release.

The last time this happened was when deploying the cluster when a rook release was ongoing, and hence the image version the script was looking for was not yet published. A similar error is what I see above (`manifest for rook/ceph:v1.6.5 not found`).

We can stick to `1.6.5` for better reliability in such cases. @hatfieldbrian let me know if you want me to modify this to a released version.

Reply:
Upsides to a "latest" version:

- if there were an bug and a pod were redeployed with a newer "latest" version it could get fixed automatically

Upsides to a fixed version:

- one fewer variable (rook version)
- script works even at times when version has changed but image hasn't been published (e.g. last two Thursdays)

I vote for using a specific version.  ocm-minikube-ramen.sh completed successfully with rook v1.6.5, so that one is fine with me.

Reply:
Done.